### ISSUE TITLE
Test failure on Windows debug mode, prologepilog-ir.mir

### ISSUE THREAD

**Issue by johnwbyrd** [llvm-mos/llvm-mos#92]
Seems to me that the vector at MOSFrameLowering.cpp:64:0 may unexpectedly be of zero length in this test.  Testing in release mode would let this issue slip by.

```
 -- Testing: 75095 tests, 8 workers --
 Testing:  0.. 10.. 20.
 FAIL: LLVM :: CodeGen/MOS/prologepilog-ir.mir (929 of 75095)
 ******************** TEST 'LLVM :: CodeGen/MOS/prologepilog-ir.mir' FAILED ********************
 Script:
 --
 : 'RUN: at line 2';   c:\git\llvm-mos\build\bin\llc.exe -mtriple=mos -run-pass=prologepilog -verify-machineinstrs -o - C:\git\llvm-mos\llvm\test\CodeGen\MOS\prologepilog-ir.mir | c:\git\llvm-mos\build\bin\filecheck.exe C:\git\llvm-mos\llvm\test\CodeGen\MOS\prologepilog-ir.mir -check-prefixes=CHECK,6502
 : 'RUN: at line 3';   c:\git\llvm-mos\build\bin\llc.exe -mtriple=mos -mcpu=mos65c02 -run-pass=prologepilog -verify-machineinstrs -o - C:\git\llvm-mos\llvm\test\CodeGen\MOS\prologepilog-ir.mir | c:\git\llvm-mos\build\bin\filecheck.exe C:\git\llvm-mos\llvm\test\CodeGen\MOS\prologepilog-ir.mir -check-prefixes=CHECK,65C02
 --
 Exit Code: 2
 
 Command Output (stdout):
 --
 $ ":" "RUN: at line 2"
 $ "c:\git\llvm-mos\build\bin\llc.exe" "-mtriple=mos" "-run-pass=prologepilog" "-verify-machineinstrs" "-o" "-" "C:\git\llvm-mos\llvm\test\CodeGen\MOS\prologepilog-ir.mir"
 # command stderr:
 PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace.
 Stack dump:
 0.	Program arguments: c:\\git\\llvm-mos\\build\\bin\\llc.exe -mtriple=mos -run-pass=prologepilog -verify-machineinstrs -o - C:\\git\\llvm-mos\\llvm\\test\\CodeGen\\MOS\\prologepilog-ir.mir
 1.	Running pass 'Function Pass Manager' on module 'C:\git\llvm-mos\llvm\test\CodeGen\MOS\prologepilog-ir.mir'.
 2.	Running pass 'Prologue/Epilogue Insertion & Frame Finalization' on function '@interrupt_csrs_save_y'
  #0 0x00007ff6d06e3809 std::_Vector_const_iterator<class std::_Vector_val<struct std::_Simple_types<class llvm::CalleeSavedInfo>>>::operator++(void) C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\vector:75:0
  #1 0x00007ff6d113f3e8 std::_Vector_iterator<class std::_Vector_val<struct std::_Simple_types<class llvm::CalleeSavedInfo>>>::operator++(void) C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\vector:284:0
  #2 0x00007ff6d22c5d30 llvm::MOSFrameLowering::assignCalleeSavedSpillSlots(class llvm::MachineFunction &, class llvm::TargetRegisterInfo const *, class std::vector<class llvm::CalleeSavedInfo, class std::allocator<class llvm::CalleeSavedInfo>> &) const C:\git\llvm-mos\llvm\lib\Target\MOS\MOSFrameLowering.cpp:64:0
  #3 0x00007ff6d113a583 assignCalleeSavedSpillSlots C:\git\llvm-mos\llvm\lib\CodeGen\PrologEpilogInserter.cpp:402:0
  #4 0x00007ff6d1137de3 `anonymous namespace'::PEI::spillCalleeSavedRegs C:\git\llvm-mos\llvm\lib\CodeGen\PrologEpilogInserter.cpp:613:0
  #5 0x00007ff6d1136fe7 `anonymous namespace'::PEI::runOnMachineFunction C:\git\llvm-mos\llvm\lib\CodeGen\PrologEpilogInserter.cpp:246:0
  #6 0x00007ff6d0478433 llvm::MachineFunctionPass::runOnFunction(class llvm::Function &) C:\git\llvm-mos\llvm\lib\CodeGen\MachineFunctionPass.cpp:72:0
  #7 0x00007ff6d00a6945 llvm::FPPassManager::runOnFunction(class llvm::Function &) C:\git\llvm-mos\llvm\lib\IR\LegacyPassManager.cpp:1439:0
  #8 0x00007ff6d00abb9e llvm::FPPassManager::runOnModule(class llvm::Module &) C:\git\llvm-mos\llvm\lib\IR\LegacyPassManager.cpp:1485:0
  #9 0x00007ff6d00a731a `anonymous namespace'::MPPassManager::runOnModule C:\git\llvm-mos\llvm\lib\IR\LegacyPassManager.cpp:1554:0
 #10 0x00007ff6d00a6de3 llvm::legacy::PassManagerImpl::run(class llvm::Module &) C:\git\llvm-mos\llvm\lib\IR\LegacyPassManager.cpp:542:0
 #11 0x00007ff6d00abe71 llvm::legacy::PassManager::run(class llvm::Module &) C:\git\llvm-mos\llvm\lib\IR\LegacyPassManager.cpp:1681:0
 #12 0x00007ff6cfe3455e compileModule C:\git\llvm-mos\llvm\tools\llc\llc.cpp:689:0
 #13 0x00007ff6cfe32978 main C:\git\llvm-mos\llvm\tools\llc\llc.cpp:388:0
 #14 0x00007ff6d35b9ac9 invoke_main d:\a01\_work\2\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:79:0
 #15 0x00007ff6d35b9bfe __scrt_common_main_seh d:\a01\_work\2\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288:0
 #16 0x00007ff6d35b9c7e __scrt_common_main d:\a01\_work\2\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:331:0
 #17 0x00007ff6d35b9c9e mainCRTStartup d:\a01\_work\2\s\src\vctools\crt\vcstartup\src\startup\exe_main.cpp:17:0
 #18 0x00007ffcf7a17034 (C:\WINDOWS\System32\KERNEL32.DLL+0x17034)
 #19 0x00007ffcf99e2651 (C:\WINDOWS\SYSTEM32\ntdll.dll+0x52651)
 
 error: command failed with exit status: 3221225477
 $ "c:\git\llvm-mos\build\bin\filecheck.exe" "C:\git\llvm-mos\llvm\test\CodeGen\MOS\prologepilog-ir.mir" "-check-prefixes=CHECK,6502"
 # command stderr:
 FileCheck error: '<stdin>' is empty.
 FileCheck command line:  c:\git\llvm-mos\build\bin\filecheck.exe C:\git\llvm-mos\llvm\test\CodeGen\MOS\prologepilog-ir.mir -check-prefixes=CHECK,6502
 
 error: command failed with exit status: 2
 ```

**Comment by mysterymath** [comment]
This should be fixed with 1ec7fd50b7d5b26f46d7d3fe04ee446d05ff09b1.

It looks like the existing code was modifying the vector while iterating through it; classic nono.

**Comment by johnwbyrd** [comment]
LGTM
